### PR TITLE
Follow up making Python tests more deterministic

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
Addressing comments in https://github.com/rapidsai/cudf/pull/17008/files#r1823318321 and https://github.com/rapidsai/cudf/pull/17008/files#r1823318898

Didn't touch the `_fuzz_testing` directory because maybe we don't want that to be deterministic?

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Review by Matt711 — APPROVED** [review]

**Review by galipremsagar — APPROVED** [review]

**Review by jameslamb — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -86,12 +86,8 @@ repos:
         types: [python]
       - id: no-unseeded-default-rng
         name: no-unseeded-default-rng
-        description: 'Enforce that no non-seeded default_rng is used and default_rng is used instead of np.random.seed'
-        entry: |
-          # Check for usage of default_rng without seeding
-          default_rng\(\)|
-          # Check for usage of np.random.seed
-          np.random.seed\(
```
Comment (vyasr) [inline .pre-commit-config.yaml:94]: Should we keep this without the parenthesis to forbid usage of np.random.seed in favor of the newer default_rng API?

**Review by vyasr — APPROVED** [review]
One suggestion for improving the future usage but it's not a blocker if we decide not to go with it (I'm not sure if we can actually do what I suggest yet).

File: .pre-commit-config.yaml (hunk 1 above)
Comment (mroeschke) [reply .pre-commit-config.yaml:94]: Ah interesting, NPY002 allows the `np.random.seed` callable to exist but complains when it's called. Sure I'll add this back.

**Pull request merged**